src/lib/operators/aggregate_sort.cpp

```diff
@@ -0,0 +1,416 @@
+#include "aggregate_sort.hpp"
+
+#include <memory>
+#include <vector>
+
+#include "aggregate/aggregate_traits.hpp"
+#include "all_type_variant.hpp"
+#include "constant_mappings.hpp"
+#include "operators/sort.hpp"
+#include "storage/segment_iterate.hpp"
+#include "table_wrapper.hpp"
+#include "type_cast.hpp"
+#include "types.hpp"
+
+namespace opossum {
+
+AggregateSort::AggregateSort(const std::shared_ptr<AbstractOperator>& in,
+                             const std::vector<AggregateColumnDefinition>& aggregates,
+                             const std::vector<ColumnID>& groupby_column_ids)
+    : AbstractAggregateOperator(in, aggregates, groupby_column_ids) {}
+
+const std::string AggregateSort::name() const { return "AggregateSort"; }
+
+template <typename ColumnType, typename AggregateType, AggregateFunction function>
+void AggregateSort::_aggregate_values(std::set<RowID>& aggregate_group_offsets, uint64_t aggregate_index,
+                                      AggregateFunctor<ColumnType, AggregateType> aggregate_function,
+                                      std::shared_ptr<const Table> sorted_table) {
+  const size_t num_groups = aggregate_group_offsets.size() + 1;
+  auto aggregate_results = std::vector<AggregateType>(num_groups);
+  auto aggregate_null_values = std::vector<bool>(num_groups, false);
+
+  uint64_t value_count = 0u;
+  uint64_t value_count_with_null = 0u;
+  std::unordered_set<ColumnType> unique_values;
+  uint64_t aggregate_group_index = 0u;
+
+  const auto& chunks = sorted_table->chunks();
+
+  std::optional<AggregateType> current_aggregate_value;
+  ChunkID current_chunk_id{0};
+  // Count*
+  if (function == AggregateFunction::Count && !_aggregates[aggregate_index].column) {
+    auto previous_group_offset = RowID{ChunkID{0u}, ChunkOffset{0}};
+    for (const auto &aggregate_group_offset : aggregate_group_offsets) {
+      if (previous_group_offset.chunk_id == aggregate_group_offset.chunk_id) {
+        value_count_with_null = aggregate_group_offset.chunk_offset - previous_group_offset.chunk_offset;
+      } else {
+        uint64_t count = 0;
+        count += chunks[previous_group_offset.chunk_id]->size() - previous_group_offset.chunk_offset;
```

Comment:
We do not want to change this or add an empty line, because all code in the else is for calculating the correct count and we think the readability is good at the moment.